chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1021 @@
+        Get the special code for a given channel.
+
+        Args:
+            channel (int): The channel number (1-199).
+
+        Returns:
+            str: The special code (8 characters long).
+        """
+        if not 1 <= channel <= 199:
+            raise ValueError("Channel number must be between 1 and 199")
+
+        special_code = self._memobj.special_codes[(channel - 1) // 4] \
+            .special_code[(channel - 1) % 4].get_raw()
+        return special_code.hex().upper()
+
+    def set_special_code(self, channel, code):
+        """
+        Set the special code for a given channel.
+
+        Args:
+            channel (int): The channel number (1-199).
+            code (str): The special code (8 characters long).
+
+        Raises:
+            ValueError: If channel is out of range or code is
+            not 8 characters long.
+        """
+        if not 1 <= channel <= 199:
+            raise ValueError("Channel number must be between 1 and 199")
+
+        if len(code) != 8:
+            raise ValueError("Special code must be exactly 8 characters long")
+
+        # Get the special code memory
+        special_code = self._memobj.special_codes[(channel - 1) // 4] \
+            .special_code[(channel - 1) % 4]
+        special_code.set_raw(bytes.fromhex(code))
+
+    # Store details about a high-level memory to the memory map
+    # This is called when a user edits a memory in the UI
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number - 1]
+
+        if mem.empty:
+            _mem.set_raw(b"\xff" * 16)
+            # Also reset all externally stored settings
+            self.set_power_level(mem.number, False)
+            self.set_scan_add_status(mem.number, False)
+            self.set_learn_code_status(mem.number, False)
+            self.set_special_code(mem.number, "FF" * 4)
+            return
+
+        self.encode_mem_blob(mem, _mem)
+
+    def _get_bool(self, setting):
+        if setting == "BATS":
+            byte1 = self._memobj.memory[3-1].global1.get_raw()
+
+            return bool(int.from_bytes(byte1, byteorder="big") & 0x08)
+        elif setting == "BEEP":
+            byte1 = self._memobj.memory[3-1].global1.get_raw()
+
+            return bool(int.from_bytes(byte1, byteorder="big") & 0x04)
+        elif setting == "WARN":
+            byte1 = self._memobj.memory[7-1].global1.get_raw()
+
+            return bool(int.from_bytes(byte1, byteorder="big") & 0x01)
+        elif setting == "SCAN":
+            byte2 = self._memobj.memory[7-1].global2.get_raw()
+
+            return bool(int.from_bytes(byte2, byteorder="big") & 0x01)
+        elif setting == "CPYC":
+            byte1 = self._memobj.memory[6-1].global3.get_raw()
+
+            return bool(int.from_bytes(byte1, byteorder="big") & 0x01)
+        elif setting == "VOXF":
+            byte1 = self._memobj.memory[4-1].global1.get_raw()
+
+            return bool(int.from_bytes(byte1, byteorder="big") & 0x01)
```

Comment:
Please don't do this sort of thing. Identify the bits in the bitwise definition and get/set them ad objects instead of doing your own bit ops.